File: 344.reverse-string.js
/*
 * @lc app=leetcode id=344 lang=javascript
 *
 * [344] Reverse String
 */

// @lc code=start
/**
 * @param {string[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function (s) 
{
    let left = 0;
    let right = s.length - 1;
    while (left < right)
    {
        [s[left], s[right]] = [s[right], s[left]];
        left++;
        right--;
    }
};
// @lc code=end

File: prev-2024/344.reverse-string.cpp
/*
 * @lc app=leetcode id=344 lang=cpp
 *
 * [344] Reverse String
 */

#include <vector>

using std::vector;

// @lc code=start
class Solution {
 public:
  void reverseString(vector<char>& s) {
    int left = 0;
    int right = s.size() - 1;

    while (left < right) {
      const char temp = s[left];
      s[left] = s[right];
      s[right] = temp;
      left++;
      right--;
    }
  }
};
// @lc code=end
